### ISSUE TITLE
tokio::task::block_in_place can only be called inside spawned tasks

### ISSUE THREAD

**Issue by Marwes** [tokio-rs/tokio-compat#19]
The following panics

```rust
#[test]
fn block_on_from_block_on() {
    let mut rt = runtime::Runtime::new().unwrap();
    rt.block_on_std(async {
        tokio_02::task::block_in_place(move || ());
    });
}
```

Easy workaround by wrapping the b`block_in_place` call with a `tokio::spawn`, but it had me confused for a bit.

**Comment by hawkw** [comment]
Yeah, this is a Tokio 0.2 limitation — there is an upstream ticket about it: https://github.com/tokio-rs/tokio/issues/1838. As per https://github.com/tokio-rs/tokio/issues/1838#issuecomment-564216581, this limitation may be removed in the future.
